web/html/src/manager/systems/virtual-list.tsx

```diff
@@ -0,0 +1,143 @@
+    <>
+      <h1>
+        <IconTag type="header-system" />
+        {t("Virtual Systems")}
+        <a
+          href={`/docs/${props.docsLocale}/reference/systems/systems-list.html`}
+          target="_blank"
+          rel="noopener noreferrer"
+        >
+          <IconTag type="header-help" />
+        </a>
+      </h1>
+
+      <div>
+        <button className="btn btn-default" onClick={addToSSM}>
+          {t("Add Selected to SSM")}
+        </button>
+      </div>
+
+      <Table
+        data="/rhn/manager/api/systems/list/virtual"
+        identifier={(item) => item.uuid}
+        initialSortColumnKey="hostServerName"
+        selectable
+        isSelectEnabled={(item) => item.hasOwnProperty("virtualSystemId")}
+        selectedItems={selectedSystems}
+        onSelect={handleSelectedSystems}
+        initialItemsPerPage={window.userPrefPageSize}
+        searchField={<SearchField filter={searchData} placeholder={t("Filter by System name: ")} />}
+        emptyText={t("No Virtual Systems.")}
+      >
+        <Column
+          columnKey="hostServerName"
+          comparator={Utils.sortByText}
+          header={t("Virtual Host")}
+          cell={(item) => {
+            return <a href={`/rhn/systems/details/Overview.do?sid=${item.hostSystemId}`}>{item.hostServerName}</a>;
+          }}
+        />
+        <Column
+          columnKey="name"
+          comparator={Utils.sortByText}
+          header={t("Virtual System")}
+          cell={(item) => {
+            if (item.systemId != null) {
+              return <a href={`/rhn/systems/details/Overview.do?sid=${item.systemId}`}>{item.name}</a>;
+            }
+            return item.name;
+          }}
+        />
+        <Column
+          columnKey="stateName"
+          comparator={Utils.sortByText}
+          header={t("Status")}
+          cell={(items) => items.stateName}
+        />
+        <Column
+          columnKey="statusType"
+          comparator={Utils.sortByText}
+          header={t("Updates")}
+          cell={(row) => {
+            if (row.statusType == null) {
+              return "";
+            }
+            return Systems.statusDisplay(row, props.isAdmin);
+          }}
+        />
+        <Column
+          columnKey="channelLabels"
+          comparator={Utils.sortByText}
+          header={t("Base Software Channel")}
+          cell={(item) => {
+            if (item.channelId != null) {
+              return <a href={`/rhn/channels/ChannelDetail.do?cid=${item.channelId}`}>{item.channelLabels}</a>;
+            }
+            return item.channelLabels;
+          }}
+        />
+      </Table>
+
+      <div className="spacewalk-csv-download">
+        <a href="/rhn/manager/systems/csv/virtualSystems" className="btn btn-link" data-senna-off="true">
+          <IconTag type="item-download-csv" />
+          Download CSV
+        </a>
+      </div>
+    </>
+  );
+}
+
+export const renderer = (id: string, docsLocale: string, isAdmin: boolean) =>
```

Comment:
Even though this is a very small bit, we usually have renderers in separate files. For an example please see any `.renderer.tsx` file.

Reply:
👍🏻